File: 58-length-of-last-word/length-of-last-word.go
func lengthOfLastWord(s string) int {
    i := len(s) - 1

    for i >= 0 && s[i] == ' ' {
        i--
    } 

    sum := 0

    for i >= 0 && s[i] != ' ' {
        i--
        sum++
    }

    return sum
}

File: 58-length-of-last-word/length-of-last-word.cpp
class Solution {
public:
    int lengthOfLastWord(string s) {
        short i = static_cast<short>(s.size()) - 1;

        while (i >= 0 && s[i] == ' ') {
            --i;
        }

        int sum = 0;

        while (i >= 0 && s[i] != ' ') {
            --i;
            ++sum;
        }

        return sum;
    }
};